### PR TITLE
ANW-451

### PR DESCRIPTION
**Pull request by avatar382**
Also includes fixes for:
ANW-175
ANW-176

### CODE REVIEW

**Comment by avatar382** [comment]
Just rebased with latest master as of 4/13.

File: backend/app/exporters/serializers/mods.rb [hunk 1]
```diff
@@ -35,7 +35,7 @@ def serialize_mods_inner(mods, xml)
 
 
     xml.language {
-      xml.languageTerm(:type => 'code') {
```
Comment (lmcglohon) [inline backend/app/exporters/serializers/mods.rb]: @avatar382 @cdibella Do we need to have both the language code and language term exported in two different XML tags? As a reference, I am looking at lines 19-27 in the spreadsheet attached to ANW-451.

File: backend/app/exporters/serializers/mods.rb [hunk 2]
```diff
@@ -143,4 +161,49 @@ def serialize_note(note, xml)
...
+      end
+    end
+    # encoding
+    attrs['encoding'] = "w3cdtf"
+
+    # keyDate
+    attrs['keyDate'] = "yes"
+
+    # point
+    attrs['point'] = "end"
```
Comment (lmcglohon) [inline backend/app/exporters/serializers/mods.rb]: @avatar382 I believe you could also have a point attribute = 'begin' as well. See comment above.

File: backend/app/exporters/serializers/mods.rb [hunk 3]
```diff
@@ -143,4 +161,49 @@ def serialize_note(note, xml)
...
+    # use date expression, if present
+    unless date['expression'].nil? || date['expression'].empty?
+      return date['expression']
+    end
+
+    # if end date specified, use begin - end form
+    unless date['end'].nil? || date['end'].empty?
+      return "#{date['begin']} - #{date['end']}"
+
+    # otherwise, just use begin date
```
Comment (lmcglohon) [inline backend/app/exporters/serializers/mods.rb]: @avatar382 @cdibella Referring to lines 39-60 in the spreadsheet, if you have a date['begin'] and a date['end'], shouldn't you have separate date tags for each along with the date['expression'] tag?

File: backend/app/exporters/serializers/mods.rb [hunk 4]
```diff
@@ -81,7 +81,7 @@ def serialize_mods_inner(mods, xml)
           xml.temporal term
         when 'uniform_title'
           xml.titleInfo term
-        when 'genre_form', 'style_period'
+        when 'genre_form', 'style_period', 'technique'
```
Comment (lmcglohon) [inline backend/app/exporters/serializers/mods.rb]: @avatar382 @cdibella From spreadsheet line 231, should this also include 'function'?

File: backend/app/exporters/models/mods.rb [hunk 5]
```diff
@@ -158,15 +165,51 @@ def handle_notes(notes)
...
+  # when the model is serialized.
+  def handle_extents_notes(notes)
+    notes.each do |note|
+      next unless note['type'] == 'physdesc' || note['type'] == 'dimensions'
+      next unless note['publish'] == true
+
+      content = ASpaceExport::Utils.extract_note_text(note)
+      mods_note = case note['type']
+                  when 'physdesc'
+                    new_mods_note('note',
```
Comment (lmcglohon) [inline backend/app/exporters/models/mods.rb]: @avatar382 @cdibella From note on line 72 of spreadsheet, shouldn't both the physdesc and dimensions notes be wrapped in a physicalDescription tag? If so, I believe that is the last (5th) input parameter for the new_mods_note method.

File: backend/app/exporters/serializers/mods.rb (hunk 1 above)
Comment (cdibella) [reply backend/app/exporters/serializers/mods.rb]: Yes, the intent is to have two different XML tags - one with a type "text" and the other with a type "code"

File: backend/app/exporters/serializers/mods.rb (hunk 4 above)
Comment (cdibella) [reply backend/app/exporters/serializers/mods.rb]: Yes, function should be in there too - all three map to the mods tag genre.

File: backend/app/exporters/models/mods.rb (hunk 5 above)
Comment (cdibella) [reply backend/app/exporters/models/mods.rb]: yes, <physicalDescription> is the wrapper tag for both of these things

Comment (cdibella) [reply backend/app/exporters/models/mods.rb]: my tag disappeared, at least on the web page, but yup, physicalDescription is the wrapper

**Comment by avatar382** [comment]
I've fixed the language code and genre fixes. Looking at the physicalDescription one, the note tags are wrapped in a physicalDescription tag, but not individually. 

Here's some sample output:
https://gist.github.com/avatar382/70a8c530fbe7760bfad12645e33c7191

Does this look right?

**Comment by lmcglohon** [comment]
@avatar382 @cdibella I think having everything wrapped inside one physical description is correct - Christine?

**Comment by cdibella** [comment]
Yes, one physical description tag, with multiple notes (depending on what’s there) inside

**Pull request merged**